src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -45,24 +45,45 @@ import scala.collection.mutable.ListBuffer
 
 
 object GroupReadsByUmi {
+  /** A type representing an actual UMI that is a string of DNA sequence. */
+  type Umi = String
+  /** A type to represent a unique ID assigned to a molecule. */
+  type MoleculeId = String
+
   private val SortPositionAttribute        = "__GRs"
   private val LibraryAttribute             = "__GRlb"
   private val InsertEndsAndStrandAttribute = "__Insert"
 
   /** Trait that can be implemented to provide a UMI assignment strategy. */
   private[umi] sealed trait UmiAssigner {
-    def assign(rawUmis: Seq[String]) : Map[String, String]
+    private val counter = new AtomicLong()
+
+    /** Take in a sequence of UMIs and assign each UMI to a unique UMI group ID. */
+    def assign(rawUmis: Seq[Umi]) : Map[Umi, MoleculeId]
```

Comment:
Nope - it's there since most of the assigners want to use the same logic, but I'm not trying to force anything since the Paired assigner can't work that same way.